client/my-sites/media-library/upload-url.jsx

```diff
@@ -13,68 +14,66 @@ import analytics from 'lib/analytics';
 import FormTextInput from 'components/forms/form-text-input';
 import MediaActions from 'lib/media/actions';
 
-export default React.createClass( {
-	displayName: 'MediaLibraryUploadUrl',
+class MediaLibraryUploadUrl extends Component {
+	static propTypes = {
+		className: PropTypes.string,
+		site: PropTypes.object,
+		onAddMedia: PropTypes.func,
+		onClose: PropTypes.func,
+	};
 
-	propTypes: {
-		site: React.PropTypes.object,
-		onAddMedia: React.PropTypes.func,
-		onClose: React.PropTypes.func,
-		className: React.PropTypes.string
-	},
+	static defaultProps = {
+		onAddMedia: noop,
+		onClose: noop,
+	};
 
-	getInitialState() {
-		return {
+	constructor() {
+		super( ...arguments );
+
+		this.state = {
 			value: '',
 			isError: false
 		};
-	},
-
-	getDefaultProps() {
-		return {
-			onAddMedia: noop,
-			onClose: noop
-		};
-	},
+	}
 
-	upload( event ) {
-		var isError = ! event.target.checkValidity();
+	upload = event => {
+		const isError = ! event.target.checkValidity();
 
-		this.setState( {
-			isError: isError
-		} );
+		this.setState( { isError } );
 
 		if ( isError || ! this.props.site ) {
 			return;
 		}
 
+		event.preventDefault();
```

Comment:
I think that if the event should be stopped, it should be done as soon as possible. Maybe it's a vestige of the old school.